modules/katamari/src/main/ts/ephox/katamari/api/Optional.ts

```diff
@@ -1,141 +1,291 @@
-import * as Fun from './Fun';
+import * as Type from './Type';
 
-/** A value which may be Optional.some(a) or Optional.none().
- *  Useful for representing partiality or pass/fail conditions that produce a value when successful.
- *  Sometimes called a "type-safe alternative to null".
- *  Can be thought of as a list of exactly zero or 1 elements.
+/**
+ * The `Optional` type represents a value (of any type) that potentially does
+ * not exist. Any `Optional<T>` can either be a `Some<T>` (in which case the
+ * value does exist) or a `None` (in which case the value does not exist). This
+ * file defines a whole lot of FP-inspired utility functions for dealing with
+ * `Optional` objects. Note: Functions are grouped based on, primarily, their
+ * Haskell roots. That said, I'm not a Haskell programmer so it might be
+ * imperfect.
```

Comment:
```suggestion
 * `Optional` objects.
 *
 * Note: Functions are grouped based on, primarily, their Haskell roots.
```
Please no personal `I'm` in this type of thing, as it's not clear who that refers to and really doesn't make sense to be in a doc string.

Reply:
I also wonder if the note should even be in this docstring at all, as it feels like it's a comment specific to the file, not to the `Optional` API.

Reply:
No longer in the doc comments. Instead, I've replaced strings like 
```
// --- Functor ---
```

with 

```
// --- Functor (name stolen from Haskell / maths) ---
```

for the names where that was appropriate.